openff/toolkit/topology/molecule.py

```diff
@@ -2444,6 +2444,26 @@ def assign_partial_charges(
         openff.toolkit.utils.toolkits.AmberToolsToolkitWrapper.assign_partial_charges
         openff.toolkit.utils.toolkits.BuiltInToolkitWrapper.assign_partial_charges
         """
+
+        # Raise a warning when users try to apply these charge methods to "large" molecules
+        WARN_LARGE_MOLECULES: Set[str] = {
+            "am1bcc",
+            "am1bccelf10",
+            "am1-mulliken",
+            "am1bccnosymspt",
+            "am1elf10",
+        }
+
+        if partial_charge_method in WARN_LARGE_MOLECULES:
+            if self.n_atoms > 150:
```

Comment:
I don't care what the threshold is set at; 100 might hit some false positives, especially with OpenEye, but too large like 300 would miss some bad inputs.